### ISSUE TITLE
Feature Request: rclone support

### ISSUE THREAD

**Issue by Jwink3101** [jedbrown/git-fat#93]
I may do the work myself but I wanted to document it.

[rclone](https://github.com/rclone/rclone) is "rsync for the cloud" and has many of the same features as rsync. I think it would be easy enough to add an rclone mode instead of rsync. Then, users can use S3, B2, etc. Basically, whatever they want!

The biggest downside to rclone over rsync is that rclone doesn't apply the rsync algorithm to only make small changes. However, the way git-fat works, I don't think it matters. git-fat pushes full copies of every file based on their sha1 so even a small change still needs a whole new copy.

I am not sure how the API would look. Probably something like if `[rclone]` is in `.gitfat` it will use rclone mode. Users would have to have rclone obviously!

**Comment by jedbrown** [comment]
Yes, I'd be interested in a way to support a range of cloud storage without git-fat needing to support each custom interface. See also #86.

**Comment by malaybasu** [comment]
This will be an awesome enhancement and would leave git-lfs out of water.
